core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -880,6 +880,12 @@ private static RexNode translateInput(RelJson relJson, int input,
       assertThat(result, isLinux(expected));
     }
 
+    RexNode betweenDoubles =
```

Comment:
I suspect this will test Doubles, but are there any tests for decimal at this point?

Reply:
Hmm you're right, there's a missing case. 
The existing test that has values (`14`, `20`) that test Integers will use BigDecimal and not Double, however 

a test case like 


```
    RexNode betweenDecimal =
        rexBuilder.makeBetween(b.literal(45),
            b.literal(BigDecimal.valueOf(20.0)),
            b.literal(BigDecimal.valueOf(30.0)));
```
results in it using Doubles ... 
Let me know if you have any ideas -- we might have to serialize the typing when creating the JSON to avoid this?